File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        l, r = 0, len(nums) - 1
        answer = [-1, -1]

        while l <= r:
            m = (l + r) // 2
            if nums[m] == target:
                answer = [m, m]
                break
            
            if nums[m] < target:
                l = m + 1
            else:
                r = m - 1

        if answer[0] > -1:
            while nums[answer[0]] == target:
                answer[0] -= 1
                if answer[0] == -1:
                    break
            answer[0] += 1 
            
            while nums[answer[1]] == target:
                answer[1] += 1
                if answer[1] == len(nums):
                    break
            answer[1] -= 1 

        return answer
            

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int start = -1;
        int end = -1;
        int index = Arrays.binarySearch(nums, target);
        if(index < 0){
            return new int[]{start, end};
        }
 
        for (int i = index; i >= 0; i--){
            if (nums[i] == target){
                start = i;
            }else{
                break;
            }
        }  

        for (int i = index; i < nums.length; i++){
            if (nums[i] == target){
                end = i;
            }else{
                break;
            }
        }
        
        return new int[]{start, end};
    }
}